pkg/ingester/transfer.go

```diff
@@ -284,7 +290,8 @@ func (i *Ingester) TransferTSDB(stream client.Ingester_TransferTSDBServer) error
 		receivedFiles.Add(float64(filesXfer))
 		level.Error(util.Logger).Log("msg", "Total files xfer", "from_ingester", fromIngesterID, "num", filesXfer)
 
-		return nil
+		// Move the tmpdir to the final location
+		return os.Rename(dir, i.cfg.TSDBConfig.Dir)
```

Comment:
From the golang doc:

> Rename renames (moves) oldpath to newpath. If newpath already exists and is not a directory, Rename replaces it.

What if the newpath (`i.cfg.TSDBConfig.Dir`) already exists and is a directory?

Reply:
I don't know what the correct behaviour should be here. If I had a leftover ingester that rejoined I would expect it to overwrite what it had with it's new data

Reply:
Does `Rename()` replace the destination if it's a non empty directory? I just would like to double check it does what we expect it to do.

Reply:
Confirmed it errors out on a directory (empty or not)

Reply:
So should it attempt to remove any preexisting directory?

Reply:
I'm dubious cause it may lead to data loss (ie. think about restarting a crashed ingester). From my perspective the ideal logic should be: OK if the destination directory not exist or empty, fail if not empty.

Reply:
Added behaviour to delete an empty dir, but throw an error on a non-empty dir.